src/test/java/com/epam/training/designpatterns/fastfoodrestaurant/ClientTest.java

```diff
@@ -1,17 +1,161 @@
 package com.epam.training.designpatterns.fastfoodrestaurant;
 
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Before;
+import org.junit.Ignore;
 import org.junit.Test;
 
-public class ClientTest {
+import com.epam.training.designpatterns.fastfoodrestaurant.entities.Client;
+import com.epam.training.designpatterns.fastfoodrestaurant.entities.Order;
+import com.epam.training.designpatterns.fastfoodrestaurant.food.Chips;
+import com.epam.training.designpatterns.fastfoodrestaurant.food.Hotdog;
+import com.epam.training.designpatterns.fastfoodrestaurant.food.Ketchup;
+import com.epam.training.designpatterns.fastfoodrestaurant.food.Mustard;
+import com.epam.training.designpatterns.fastfoodrestaurant.workstations.Chef;
+import com.epam.training.designpatterns.fastfoodrestaurant.workstations.DeliveryQueue;
+import com.epam.training.designpatterns.fastfoodrestaurant.workstations.OrderQueue;
+import com.epam.training.designpatterns.fastfoodrestaurant.workstations.Waiter;
 
+public class ClientTest {
```

Comment:
Nagyon szép, olvasható, megfelelő granularitású tesztek. Egy dolog: ezek integrációs tesztek, nem unit tesztek, mivel end-to-end letesztelik az egész rendszert valós komponensekkel. Ha a `Client` (és a többi osztály) unit tesztjeit akarjuk megírni, akkor ki kell mockolnunk a `Client` minden kollaborátorát és preparált válaszokat kell adnunk a hívásokra.

Toolok: JUnit, TestNG, EasyMock, Mockito, Hamcrest, stb.

Javaslom, hogy ezt az osztályt nevezd át `IntegrationTest`re és írj pár unit tesztet is gyakorlásképp!

Reply:
Igen, teljesen igaz. Fogok hozzá írni Mockito/JUnit párossal unit testeket is. Ezt pedig átnevezem integration test-re, és a maven lifecycle-ben is átállítom hogy verify során fusson.